src/Doppler.sol

```diff
@@ -758,11 +764,13 @@ contract Doppler is BaseHook {
         uint256 assetAvailable;
 
         if (isToken0) {
-            numeraireAvailable = uint256(uint128(positionDeltas.amount1())) - uint128(feeDeltas.amount1());
+            numeraireAvailable = uint256(uint128(positionDeltas.amount1())) + key.currency1.balanceOfSelf()
+                - uint128(state.feesAccrued.amount1());
             assetAvailable = uint256(uint128(positionDeltas.amount0())) + key.currency0.balanceOfSelf()
                 - uint128(feeDeltas.amount0());
         } else {
-            numeraireAvailable = uint256(uint128(positionDeltas.amount0())) - uint128(feeDeltas.amount0());
+            numeraireAvailable = uint256(uint128(positionDeltas.amount0())) + key.currency0.balanceOfSelf()
+                - uint128(state.feesAccrued.amount0());
```

Comment:
just thinking out loud. im pretty sure there's no underflow risk here but it came to mind, feesAccrued should never exceed balanceOfSelf. even if we added a check that reverts when it does exceed, the underflow would already revert so no use in adding an explicit check.